lib/IDE/SourceEntityWalker.cpp

```diff
@@ -422,6 +426,34 @@ std::pair<bool, Expr *> SemaAnnotator::walkToExprPre(Expr *E) {
     if (!walkToExprPost(E))
       return { false, nullptr };
     return { false, E };
+  } else if (auto OEE = dyn_cast<OpenExistentialExpr>(E)) {
+    // Record opaque value.
+    OpaqueValueMap[OEE->getOpaqueValue()] = OEE->getExistentialValue();
+
+    if (!OEE->getSubExpr()->walk(*this))
+      return { false, nullptr };
+    if (!walkToExprPost(E))
+      return { false, nullptr };
+    return { false, E };
+  } else if (auto CUCE = dyn_cast<CollectionUpcastConversionExpr>(E)) {
+    // Ignore conversion expressions. We don't handle OpaqueValueExpr here
+    // because it's only in conversion expressions. Instead, just walk into
+    // sub expression.
+    if (!CUCE->getSubExpr()->walk(*this))
+      return { false, nullptr };
+    if (!walkToExprPost(E))
+      return { false, nullptr };
+    return { false, E };
+  } else if (auto OVE = dyn_cast<OpaqueValueExpr>(E)) {
+    // Walk into mapped value.
+    if (auto *expr = OpaqueValueMap[OVE]) {
+      OpaqueValueMap.erase(OVE);
```

Comment:
This will leave `NULL` entries in the `OpaqueValueMap`, if in fact `OVE` isn't there... but it should always be there. Perhaps `assert` that it's always there, or use `find` if you want to be defensive?

Rather than erasing here---which will break if at some later point we start allowing the `OpaqueValueExpr` to show up more than one---I recommend you take the suggestion from my previous comment and erase after walking the `OpenExistentialExpr`'s subexpression.

Reply:
> This will leave NULL entries in the OpaqueValueMap, if in fact OVE isn't there...

I didn't know this. Fixed to use `find`.
I'd like to be defensive here because, in future, other expression kind might start to use `OpaqueValueExpr` without modifying this special handling logic.